soh/soh/ResourceManagerHelpers.cpp

```diff
@@ -503,3 +512,139 @@ extern "C" void ResourceMgr_ClearSkeletons() {
 extern "C" s32* ResourceMgr_LoadCSByName(const char* path) {
     return (s32*)ResourceMgr_GetResourceDataByNameHandlingMQ(path);
 }
+
+bool IsSharedScene(int16_t sceneNum) {
+    return sceneNum > SCENE_INSIDE_GANONS_CASTLE || sceneNum == SCENE_GANONS_TOWER || sceneNum == SCENE_THIEVES_HIDEOUT;
+}
+
+std::string GetScenePath(int16_t sceneNum) {
+    std::string sceneName = gSceneTable[sceneNum].sceneFile.fileName;
+    std::string path = "alt/scenes/shared/" + sceneName + "/*";
+    if (!IsSharedScene(sceneNum)) {
+        size_t pos = 0;
+        if ((pos = path.find("/shared/", 0)) != std::string::npos) {
```

Comment:
just curious here, why do we need to check the path for `/shared/` when we've already checked `!IsSharedScene`

Reply:
It's replacing shared. The original starts with shared in it, this finds the position of it to insert mq or nonmq. I don't know why it checks for npos there, I copied it from your WIP XD

Reply:
i wish i could ask my past self why this was in there, because current me definitely doesn't know